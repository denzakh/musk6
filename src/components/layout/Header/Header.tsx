import { FC } from 'react'
import HeaderBg from './header.jpg'
import { Nav } from '../Nav'

interface HeaderProps {
  cartCount: number
}

export const Header: FC<HeaderProps> = ({ cartCount }) => {
  return (
    <header
      className="bg-cover bg-no-repeat bg-[#2a59bd]"
      style={{ backgroundImage: `URL(${HeaderBg})` }}
    >
      <div className="backdrop-brightness-70 md:backdrop-brightness-100">
        <div className="container">
          <div className="flex items-center text-white min-h-[45vw] lg:min-h-[380px] pl-2 md:pl-0">
            <div className="flex flex-col gap-2 md:items-center pt-4 ">
              <div className="text-3xl md:text-5xl font-header font-bold drop-shadow-lg pl-2 md:pl-0">
                Star Jedi
              </div>
              <div className="drop-shadow-sm">
                tienda de productos <br className="md:hidden" />
                exclusivos
              </div>
            </div>

            <div className="grow"></div>
            <div className=""></div>
          </div>
        </div>
      </div>
      <div className="bg-[rgba(0,0,0,0.5)]" style={{ backdropFilter: 'blur(5px)' }}>
        <div className="container">
          <Nav cartCount={cartCount} />
        </div>
      </div>
    </header>
  )
}
