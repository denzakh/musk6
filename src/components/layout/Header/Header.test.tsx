import { describe, it, expect } from 'vitest'
import { render } from '@testing-library/react'
import { Header } from './Header'
import { BrowserRouter } from 'react-router'

describe('Header', async () => {
  it('Si no hay productos, no se muestra el nÃºmero de productos en el carrito.', async () => {
    const comp = render(
      <BrowserRouter>
        <Header cartCount={0} />
      </BrowserRouter>
    )

    expect(comp).toMatchSnapshot()
  })
})
