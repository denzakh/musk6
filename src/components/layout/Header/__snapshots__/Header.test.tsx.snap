// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > Si no hay productos, no se muestra el número de productos en el carrito. 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="bg-cover bg-no-repeat bg-[#2a59bd]"
      >
        <div
          class="backdrop-brightness-70 md:backdrop-brightness-100"
        >
          <div
            class="container"
          >
            <div
              class="flex items-center text-white min-h-[45vw] lg:min-h-[380px] pl-2 md:pl-0"
            >
              <div
                class="flex flex-col gap-2 md:items-center pt-4 "
              >
                <div
                  class="text-3xl md:text-5xl font-header font-bold drop-shadow-lg pl-2 md:pl-0"
                >
                  Star Jedi
                </div>
                <div
                  class="drop-shadow-sm"
                >
                  tienda de productos 
                  <br
                    class="md:hidden"
                  />
                  exclusivos
                </div>
              </div>
              <div
                class="grow"
              />
              <div
                class=""
              />
            </div>
          </div>
        </div>
        <div
          class="bg-[rgba(0,0,0,0.5)]"
        >
          <div
            class="container"
          >
            <div
              class=""
            >
              <div
                class="min-h-[80px] p-4 lg:flex justify-center items-center gap-x-10 hidden"
              >
                <a
                  class="text-white text-2xl hover:text-cyan-300"
                  data-discover="true"
                  href="/"
                >
                  Inicio
                </a>
                <a
                  class="text-white text-2xl hover:text-cyan-300"
                  data-discover="true"
                  href="/"
                >
                  Libros
                </a>
                <a
                  class="text-white text-2xl hover:text-cyan-300"
                  data-discover="true"
                  href="/"
                >
                  Películas
                </a>
                <a
                  class="text-white text-2xl hover:text-cyan-300"
                  data-discover="true"
                  href="/"
                >
                  Cómics
                </a>
                <a
                  class="text-white text-2xl hover:text-cyan-300"
                  data-discover="true"
                  href="/"
                >
                  Juguetes
                </a>
                <a
                  class="text-white text-2xl hover:text-cyan-300"
                  data-discover="true"
                  href="/"
                >
                  Otros
                </a>
                <a
                  class="text-white text-2xl hover:text-cyan-300 flex items-center gap-4"
                  data-discover="true"
                  href="/cart"
                >
                  <div>
                    Carrito de compras
                  </div>
                </a>
              </div>
              <div
                class="flex justify-between items-center lg:hidden"
              >
                <details
                  class="dropdown"
                >
                  <summary
                    class="btn m-1"
                  >
                    Menu
                  </summary>
                  <ul
                    class="menu dropdown-content bg-base-100 rounded-box z-1 w-52 p-2 shadow-sm"
                  >
                    <li
                      class="mt-2"
                    >
                      <a
                        class=""
                        data-discover="true"
                        href="/"
                      >
                        Inicio
                      </a>
                    </li>
                    <li
                      class="mt-2"
                    >
                      <a
                        class=""
                        data-discover="true"
                        href="/"
                      >
                        Libros
                      </a>
                    </li>
                    <li
                      class="mt-2"
                    >
                      <a
                        class=""
                        data-discover="true"
                        href="/"
                      >
                        Películas
                      </a>
                    </li>
                    <li
                      class="mt-2"
                    >
                      <a
                        class=""
                        data-discover="true"
                        href="/"
                      >
                        Cómics
                      </a>
                    </li>
                    <li
                      class="mt-2"
                    >
                      <a
                        class=""
                        data-discover="true"
                        href="/"
                      >
                        Juguetes
                      </a>
                    </li>
                    <li
                      class="mt-2"
                    >
                      <a
                        class=""
                        data-discover="true"
                        href="/"
                      >
                        Otros
                      </a>
                    </li>
                  </ul>
                </details>
                <div>
                  <a
                    class="text-white text-xl hover:text-cyan-300 flex items-center"
                    data-discover="true"
                    data-testid="cartBtn"
                    href="/cart"
                  >
                    <div
                      class="pr-2"
                    >
                      Carrito
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="bg-cover bg-no-repeat bg-[#2a59bd]"
    >
      <div
        class="backdrop-brightness-70 md:backdrop-brightness-100"
      >
        <div
          class="container"
        >
          <div
            class="flex items-center text-white min-h-[45vw] lg:min-h-[380px] pl-2 md:pl-0"
          >
            <div
              class="flex flex-col gap-2 md:items-center pt-4 "
            >
              <div
                class="text-3xl md:text-5xl font-header font-bold drop-shadow-lg pl-2 md:pl-0"
              >
                Star Jedi
              </div>
              <div
                class="drop-shadow-sm"
              >
                tienda de productos 
                <br
                  class="md:hidden"
                />
                exclusivos
              </div>
            </div>
            <div
              class="grow"
            />
            <div
              class=""
            />
          </div>
        </div>
      </div>
      <div
        class="bg-[rgba(0,0,0,0.5)]"
      >
        <div
          class="container"
        >
          <div
            class=""
          >
            <div
              class="min-h-[80px] p-4 lg:flex justify-center items-center gap-x-10 hidden"
            >
              <a
                class="text-white text-2xl hover:text-cyan-300"
                data-discover="true"
                href="/"
              >
                Inicio
              </a>
              <a
                class="text-white text-2xl hover:text-cyan-300"
                data-discover="true"
                href="/"
              >
                Libros
              </a>
              <a
                class="text-white text-2xl hover:text-cyan-300"
                data-discover="true"
                href="/"
              >
                Películas
              </a>
              <a
                class="text-white text-2xl hover:text-cyan-300"
                data-discover="true"
                href="/"
              >
                Cómics
              </a>
              <a
                class="text-white text-2xl hover:text-cyan-300"
                data-discover="true"
                href="/"
              >
                Juguetes
              </a>
              <a
                class="text-white text-2xl hover:text-cyan-300"
                data-discover="true"
                href="/"
              >
                Otros
              </a>
              <a
                class="text-white text-2xl hover:text-cyan-300 flex items-center gap-4"
                data-discover="true"
                href="/cart"
              >
                <div>
                  Carrito de compras
                </div>
              </a>
            </div>
            <div
              class="flex justify-between items-center lg:hidden"
            >
              <details
                class="dropdown"
              >
                <summary
                  class="btn m-1"
                >
                  Menu
                </summary>
                <ul
                  class="menu dropdown-content bg-base-100 rounded-box z-1 w-52 p-2 shadow-sm"
                >
                  <li
                    class="mt-2"
                  >
                    <a
                      class=""
                      data-discover="true"
                      href="/"
                    >
                      Inicio
                    </a>
                  </li>
                  <li
                    class="mt-2"
                  >
                    <a
                      class=""
                      data-discover="true"
                      href="/"
                    >
                      Libros
                    </a>
                  </li>
                  <li
                    class="mt-2"
                  >
                    <a
                      class=""
                      data-discover="true"
                      href="/"
                    >
                      Películas
                    </a>
                  </li>
                  <li
                    class="mt-2"
                  >
                    <a
                      class=""
                      data-discover="true"
                      href="/"
                    >
                      Cómics
                    </a>
                  </li>
                  <li
                    class="mt-2"
                  >
                    <a
                      class=""
                      data-discover="true"
                      href="/"
                    >
                      Juguetes
                    </a>
                  </li>
                  <li
                    class="mt-2"
                  >
                    <a
                      class=""
                      data-discover="true"
                      href="/"
                    >
                      Otros
                    </a>
                  </li>
                </ul>
              </details>
              <div>
                <a
                  class="text-white text-xl hover:text-cyan-300 flex items-center"
                  data-discover="true"
                  data-testid="cartBtn"
                  href="/cart"
                >
                  <div
                    class="pr-2"
                  >
                    Carrito
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
